lib/core/src/lib/dialogs/unsaved-changes-dialog/unsaved-changes.guard.spec.ts

```diff
@@ -43,63 +75,70 @@ describe('UnsavedChangesGuard', () => {
+                afterClosed$.next(true);
             });
-            afterClosed$.next(true);
-        });
 
-        it('should return false when unsaved is set to true and result of dialog is false', (done) => {
-            guard.unsaved = true;
+            it('should set unsaved to false when unsaved is set to true and result of dialog is true', (done) => {
+                guard.unsaved = true;
 
-            (guard.canDeactivate() as Observable<boolean>).subscribe((allowed) => {
-                expect(allowed).toBeFalse();
-                done();
+                expectGuardToBe(false, done, true);
+                afterClosed$.next(true);
             });
-            afterClosed$.next(false);
-        });
 
-        it('should keep unsaved set to true when unsaved was to true and result of dialog is false', (done) => {
-            guard.unsaved = true;
+            it('should return false if afterClosed subject value was undefined', (done) => {
+                guard.unsaved = true;
 
-            (guard.canDeactivate() as Observable<boolean>).subscribe(() => {
-                expect(guard.unsaved).toBeTrue();
-                done();
+                expectGuardToBe(false, done);
+                afterClosed$.next(undefined);
             });
-            afterClosed$.next(false);
-        });
 
-        it('should set unsaved to false when unsaved is set to true and result of dialog is true', (done) => {
-            guard.unsaved = true;
+            it('should return false when unsaved is set to true and result of dialog is false', (done) => {
+                guard.unsaved = true;
 
-            (guard.canDeactivate() as Observable<boolean>).subscribe(() => {
-                expect(guard.unsaved).toBeFalse();
-                done();
+                expectGuardToBe(false, done);
+                afterClosed$.next(false);
+            });
+
+            it('should keep unsaved set to true when unsaved was to true and result of dialog is false', (done) => {
+                guard.unsaved = true;
+
+                expectGuardToBe(true, done, true);
+                afterClosed$.next(false);
             });
-            afterClosed$.next(true);
         });
 
-        it('should return false if afterClosed subject value was undefined', (done) => {
-            guard.unsaved = true;
+        describe('Without auth', () => {
+            it('should return true when user is logged out of authenticationService', (done) => {
+                spyOn(authService, 'isLoggedIn').and.returnValue(false);
 
-            (guard.canDeactivate() as Observable<boolean>).subscribe((result) => {
-                expect(result).toBeFalse();
-                done();
+                expectGuardToBe(true, done);
+                afterClosed$.next(true);
+            });
+
+            it('should return true when authGuardBaseService.withCredentials returns true', (done) => {
+                spyOn(appConfigService, 'get').and.returnValue(false);
```

Comment:
We should avoid mocking inner dependencies. Instead, please mock out withCredentials property from AuthGuardService which is used by UnsavedChangesGuard. It would look something like: 
```ts
spyOnProperty(authGuardService, 'withCredentials').and.returnValue(false);
```

Please note I used spyOnProperty to mock value returned by withCredentials getter.